internal/pkg/iface/iface.go

```diff
@@ -64,8 +65,17 @@ func FirstPublicAddress() (string, error) {
 	}
 	ipv6addr := ""
 	for _, i := range ifs {
-		if i.Name == "vxlan.calico" || i.Name == "kube-bridge" {
-			// Skip calico and kube-router interfaces
+		switch {
+		// Skip CNI related interfaces and the k0s VIP interface
+		case i.Name == "vxlan.calico":
+			continue
+		case i.Name == "kube-bridge":
+			continue
+		case i.Name == "dummyvip0":
```

Comment:
What's this? :sweat_smile: I guess this is keepalived? Might be good to add comments for each entry where they're from. (Ideally with some links to upstream docs or sources :see_no_evil:)

Reply:
Fixed, PTAL